pkg/monitortests/node/nodestateanalyzer/monitortest.go

```diff
@@ -34,6 +34,14 @@ func (*nodeStateAnalyzer) ConstructComputedIntervals(ctx context.Context, starti
 }
 
 func (*nodeStateAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	// Fail tests when monitor test flags this as an error
+	for _, val := range finalIntervals {
+		if val.Level.String() == "Error" {
```

Comment:
I don't think this will do what you expect.  `finalIntervals` is all the intervals from every monitortest.  Many of them will choose `Error` for their level.  I think you also need to check for the locator being a node and the reason being  `NotReadyUnexpected`.

Reply:
Thanks for your help on that. I'll work on that.

Reply:
pushed an update. PTAL.